src/main/java/com/gluonhq/substrate/model/ProjectConfiguration.java

```diff
@@ -268,7 +268,16 @@ public void setTarget(Triplet targetTriplet) {
         this.targetTriplet = targetTriplet;
     }
 
-    public Triplet getHostTriplet() {
+    /**
+     * Retrieve the host triplet for this configuration.
+     * The host triplet is always the triplet for the current runtime, e.g. it should not be set (apart for testing)
+     * @return the Triplet for the current executing host
+     * @throws IllegalArgumentException in case the current operating system is not supported
+     */
+    public Triplet getHostTriplet() throws IllegalArgumentException {
+        if (hostTriplet == null) {
```

Comment:
?: would be cleaner here IMO

Reply:
We need to set the hostTriplet if not yet done, so a ? : would need to do a set + return and that seems complex?

Reply:
you should be able to do ` return hostTriplet = hostTriplet != null? Triplet.fromCurrentOS() : null;`

Reply:
that would keep hostTriplet null in case it was null